### PR TITLE
Update project and add shouldFail()

### LINKED ISSUE #2: How to add t.shouldFail()?
https://github.com/spion/blue-tape
`t.shouldFail(promise P, optional class C)`

Seems like I have to replicate most of the `tape` interface. Any better idea?

### PR DESCRIPTION
**Pull request by unional**
Add test and passed. :rose:

Fixes #2

### CODE REVIEW

**Comment by unional** [comment]
Yey

**Comment by blakeembrey** [comment]
@unional Can you please allow some time for people to review? You've merged and released a bug 👎

File: index.d.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
 import tape = require('tape');
 
+declare module 'tape' {
+  interface Test {
+    shouldFail(p: PromiseLike<any>, c?: new() => Error): PromiseLike<any>;
```
Comment (blakeembrey) [inline index.d.ts:5]: Please import the `es6-promise` library. This interface relies on a global from TypeScript that is **only** available in ES6+.

**Comment by unional** [comment]
Oops!

**Comment by blakeembrey** [comment]
It's ok, easy mistake. I only just got around to reviewing, but now you know why I add `es6-promise` as a dependency regularly 😄

**Comment by unional** [comment]
Yes Sir! :stuck_out_tongue:

**Comment by unional** [comment]
Use the `Thenable`?

``` js
import Promise = require('es6-promise');
import tape = require('tape');

declare module 'tape' {
  interface Test {
    shouldFail(p: Promise.Thenable<any>, c?: new() => Error): Promise.Thenable<any>;
  }
}

export = tape;

```

**Comment by unional** [comment]
Or 

``` js
import {Promise} from 'es6-promise';
import tape = require('tape');

declare module 'tape' {
  interface Test {
    shouldFail(p: Promise<any>, c?: new() => Error): Promise<any>;
  }
}

export = tape;
```

**Comment by blakeembrey** [comment]
The `Thenable` should be plenty here 👍 That's all that's required structurally.

**Comment by unional** [comment]
Done https://github.com/typed-typings/npm-blue-tape/pull/5

**Comment by unional** [comment]
@blakeembrey does other types exposed by TS has the same issue? Should I set `tsconfig.json/compilerOptions/noLib` to `false` for all typings?

**Comment by blakeembrey** [comment]
> does other types exposed by TS has the same issue?

I don't think `noLib` is a great fix. Basically, use the environment you expect the type definition to be used in. If we're building `node.d.ts`, yes, we should use `noLib` and use `lib.core.d.ts` instead of remove the browser added types. If we're building anything else, probably set the target to be ES5, which would catch the `Promise` issue (because it's defined in `lib.es6.d.ts`).

**Pull request merged**